sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java

```diff
@@ -117,507 +72,25 @@ public Mono<HttpResponse> send(HttpRequest request, Context contextData) {
-     *
-     * 'disallowed status code' is one of the status code defined in the provided SwaggerMethodParser or is in the int[]
-     * of additional allowed status codes.
-     *
-     * @param asyncDecodedResponse The HttpResponse to check.
-     * @param methodParser The method parser that contains information about the service interface method that initiated
-     * the HTTP request.
-     * @param options Additional options passed as part of the request.
-     * @return An async-version of the provided decodedResponse.
-     */
-    private static Mono<HttpDecodedResponse> ensureExpectedStatus(final Mono<HttpDecodedResponse> asyncDecodedResponse,
-        final SwaggerMethodParser methodParser, RequestOptions options) {
-        return asyncDecodedResponse.flatMap(decodedResponse -> {
-            int responseStatusCode = decodedResponse.getSourceResponse().getStatusCode();
+        RestProxyUtils.validateResumeOperationIsNotPresent(method);
 
-            // If the response was success or configured to not return an error status when the request fails,
-            // return the decoded response.
-            if (methodParser.isExpectedResponseStatusCode(responseStatusCode)
-                || (options != null && options.getErrorOptions().contains(ErrorOptions.NO_THROW))) {
-                return Mono.just(decodedResponse);
-            }
-
-            // Otherwise, the response wasn't successful and the error object needs to be parsed.
-            //
-            // First, try to create an error with the response body.
-            // If there is no response body create an error without the response body.
-            // Finally, return the error reactively.
-            return decodedResponse.getSourceResponse().getBodyAsByteArray()
-                .map(bytes -> instantiateUnexpectedException(methodParser.getUnexpectedException(responseStatusCode),
-                    decodedResponse.getSourceResponse(), bytes, decodedResponse.getDecodedBody(bytes)))
-                .switchIfEmpty(Mono.fromSupplier(() -> instantiateUnexpectedException(
-                    methodParser.getUnexpectedException(responseStatusCode), decodedResponse.getSourceResponse(),
-                    null, null)))
-                .flatMap(Mono::error);
-        });
-    }
+        // Note: request options need to be evaluated here, as it is a public class with package private methods.
+        // Evaluating here allows the package private methods to be invoked here for downstream use.
+        final SwaggerMethodParser methodParser = getMethodParser(method);
+        RequestOptions options = methodParser.setRequestOptions(args);
+        Context context = methodParser.setContext(args);
+        boolean isReactive = methodParser.isReactive();
+        boolean isStreamResponseType = methodParser.isStreamResponse();
+        boolean syncRestProxyEnabled = Contexts.with(context).isRestProxySyncProxyEnabled();
```

Comment:
Let's not add this to public API surface.
This switch will be eventually taken away. We should use "magic private string constant" in places we need this.